lib/dal/src/secret.rs

```diff
@@ -70,33 +87,120 @@ pk!(SecretId);
 /// therefore safe to expose via external API.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct Secret {
-    pk: SecretPk,
     id: SecretId,
-    name: String,
+
+    // TODO(nick): evaluate how these three fields will work with the new engine.
+    #[serde(flatten)]
+    timestamp: Timestamp,
+    created_by: Option<UserPk>,
+    updated_by: Option<UserPk>,
+
+    pk: SecretPk,
     key_pair_pk: KeyPairPk,
+    name: String,
     definition: String,
     description: Option<String>,
-    #[serde(flatten)]
-    tenancy: Tenancy,
-    #[serde(flatten)]
+}
+
+#[derive(EnumDiscriminants, Serialize, Deserialize, PartialEq)]
+pub enum SecretContent {
+    V1(SecretContentV1),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct SecretContentV1 {
     timestamp: Timestamp,
     created_by: Option<UserPk>,
     updated_by: Option<UserPk>,
-    #[serde(flatten)]
-    visibility: Visibility,
-}
 
-impl_standard_model! {
-    model: Secret,
     pk: SecretPk,
-    id: SecretId,
-    table_name: "secrets",
-    history_event_label_base: "secret",
-    history_event_message_name: "Secret"
+    key_pair_pk: KeyPairPk,
+    name: String,
+    definition: String,
+    description: Option<String>,
+}
+
+impl From<Secret> for SecretContentV1 {
+    fn from(value: Secret) -> Self {
+        Self {
+            timestamp: value.timestamp,
+            created_by: value.created_by,
+            updated_by: value.updated_by,
+            pk: value.pk,
+            key_pair_pk: value.key_pair_pk,
+            name: value.name,
+            definition: value.definition,
+            description: value.description,
+        }
+    }
 }
 
 impl Secret {
-    standard_model_accessor_ro!(name, str);
+    pub fn assemble(id: SecretId, inner: SecretContentV1) -> Self {
+        Self {
+            id,
+            timestamp: inner.timestamp,
+            created_by: inner.created_by,
+            updated_by: inner.updated_by,
+            pk: inner.pk,
+            key_pair_pk: inner.key_pair_pk,
+            name: inner.name,
+            definition: inner.definition,
+            description: inner.description,
+        }
+    }
+
+    // TODO(nick): to maintain API compatibility with main, we need "EncryptedSecret::new" to create
+    // this. We may want the opposite to happen in the future. We should decide this after the
+    // switchover. Let's consume the object for now to help ensure the underlying "encrypted secret"
+    // is not used where it shouldn't be.
+    pub async fn new(
+        ctx: &DalContext,
+        new_encrypted_secret: EncryptedSecret,
+    ) -> SecretResult<Self> {
```

Comment:
This is likely backwards and/or the wrong shape in the long run. However, let's satisfy existing usage patterns first before refining.